src/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -575,6 +575,49 @@ public bool Remove(TKey key)
             return false;
         }
 
+        public bool Remove(TKey key, out TValue value)
+        {
+            if (key == null)
+            {
+                value = default(TValue);
```

Comment:
Also, don't forget to remove this assignment as if we are throwing an Exception it doesn't make sense to change the out parameter value.